cpp/frontend_pyreadline.cc

```diff
@@ -4,73 +4,235 @@
 
 #include "_build/detected-cpp-config.h"
 
+#if HAVE_READLINE
+  #include <readline/history.h>
+  #include <readline/readline.h>
+#endif
+
 namespace py_readline {
 
+static Readline* gReadline = nullptr;
+
+// Assuming readline 4.0+
+#if HAVE_READLINE
+
+  #define completion_matches(x, y) \
```

Comment:
Can we just remove the macro?  It seems like it's only used in one place or so



I have been trying to use CAPS() for macros consistently too

Reply:
ah yeah. this isn't really necessary. i guess `line_input` had it because of the cross-platform stuff